src/realm.js

```diff
@@ -96,6 +97,9 @@ export class Effects {
 
     this.canBeApplied = true;
     this._id = effects_uid++;
+    if (priorEffects) {
+      this.priorEffects = priorEffects;
```

Comment:
Either always assign, or change field declaration to `priorEffects?: Effects`.

Reply:
That is not legal syntax for properties and I (and you) prefer not to make properties be of type ?Effects.